sql/parser/type_check.go

```diff
@@ -228,6 +231,63 @@ func typeCheckComparisonOp(op ComparisonOp, dummyLeft, dummyRight Datum) (Datum,
 		dummyLeft.Type(), op, dummyRight.Type())
 }
 
+func typeCheckConditionalExpr(expr *ConditionalExpr) (Datum, error) {
+	switch expr.Operator {
+	case If:
+		if len(expr.Exprs) != 3 {
```

Comment:
we can do these checks statically. let's make ConditionalExpr an interface and give each of these a type